# Blatt 8
Nikola Mang, Salem Rezik, Noah Scheugenpflug

## Aufgabe 16

### a)

Die Definition der bedingten Wahrscheinlichkeit ist

\begin{equation}
    \frac{p\left(A, B\right)}{p\left(B\right)} = p\left(A|B\right)\;.
\end{equation}

Daraus folgt

\begin{equation}
     p(A|B)\cdot p(B) = p(B|A)\cdot p(A)\;.
\end{equation}

Dividieren der Gleichung mit $p(A)$ liefert den Satz von Bayes

\begin{equation}
    \frac{p(A|B)\cdot p(B)}{p(A)} = p(B|A)
\end{equation}

### b)

In [28]:
import numpy as np
import matplotlib.pyplot as plt

In [29]:
wscj = 6
wstj = 3
fhj  = 3
fnj  = 6
thj  = 0
tmj  = 6
tkj  = 3
asj  = 2
abj  = 4
arj  = 3
wscn = 2
wstn = 3
fhn  = 4
fnn  = 1
thn  = 1
tmn  = 1
tkn  = 3
asn  = 1
abn  = 1
arn  = 3

Berechnen der bedingten Wahrscheinlichkeit für Fußball berechnet durch $\frac{\text{# Werte ja}}{\text{# Werte gesamt}}$.

In [30]:
pwst = wstj / (wstj + wstn)
pwsc = wscj / (wscj + wscn)
pfh  = fhj  / (fhj  + fhn)
pfn  = fnj  / (fnj  + fnn)
pth  = thj  / (thj  + thn)
ptm  = tmj  / (tmj  + tmn)
ptk  = tkj  / (tkj  + tkn)
pas  = asj  / (asj  + asn)
pab  = abj  / (abj  + abn)
par  = arj  / (arj  + arn)

Die Wahrscheinlichkeit, dass heute Fußball gespielt wird bei den gegebenen Bedingungen: Wind - stark, Feuchtigkeit - hoch, Temperatur - kalt, Ausblick - sonnig; wird durch das Produkt der bedingten Wahrscheinlichkeiten der gegebenen Bedingungen berechnet.

In [31]:
phf = pwst * pfh * ptk * pas
phf

0.07142857142857142

Damit ergibt sich die Wahrscheinlichkeit, dass heute Fußball gespielt wird zu $7.14\%$.

### c)

Es werden zuerst die Schritte wie in b) nur mit anderen Bedingungen: Wind - schwach, Feuchtigkeit - hoch, Temperatur - heiß, Ausblick - sonnig; wiederholt.

In [32]:
pmf = pwsc * pfh * pth * pas
pmf

0.0

Da die bedingte Wahrscheinlichkeit für eine hohe Temperatur $0\%$ ist, da in dem Datensatz keine heiße Temperatur gemessen wurde, wenn Fußball gespielt wurde, und die Gesamtwahrscheinlichkeit ein Produkt der Wahrscheinlichkeiten ist, ergibt sich diese ebenfalls zu $0\%$.

In [33]:
pmf2 = pwsc * pfh * pas
pmf2

0.21428571428571425

Dieses Problem wird umgangen, in dem die Wahrscheinlichkeit für eine heiße Temperatur ignoriert wird und nicht mit multipliziert wird. Damit ergibt sich die Wahrscheinlichkeit, dass morgen Fußball gespielt wird zu $21,43\%$.